packages/amplify-graphql-generation-transformer/src/grapqhl-generation-transformer.ts

```diff
@@ -0,0 +1,219 @@
+
+  generateResolvers = (ctx: TransformerContextProvider): void => {
+    if (this.directives.length === 0) return;
+
+    this.directives.forEach((directive) => {
+      const { parent, field } = directive;
+      const fieldName = field.name.value;
+      const parentName = parent.name.value;
+
+      const directiveWithToolConfig: GenerationConfigurationWithToolConfig = {
+        ...directive,
+        toolConfig: createResponseTypeTool(field, ctx),
+      };
+
+      const stackName = `Generation${this.capitalizeFirstLetter(fieldName)}BedrockDataSourceStack`;
+      const stack = this.createStack(ctx, stackName);
+
+      const resolverResourceId = ResolverResourceIDs.ResolverResourceID(parentName, fieldName);
+      const httpDataSourceId = HttpResourceIDs.HttpDataSourceID(`GenerationBedrockDataSource-${fieldName}`);
+      const dataSource = this.createBedrockDataSource(ctx, directive, stack.region, stackName, httpDataSourceId);
+      const invokeBedrockFunction = createInvokeBedrockResolverFunction(directiveWithToolConfig);
+
+      this.createPipelineResolver(ctx, parentName, fieldName, resolverResourceId, invokeBedrockFunction, dataSource);
+    });
+  };
+
+  private capitalizeFirstLetter(str: string): string {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  }
+
+  /**
+   * Creates a new CDK stack for the Generation transformer.
+   * @param {TransformerContextProvider} ctx - The transformer context provider.
+   * @param {string} stackName - The name of the stack to create.
+   * @returns {cdk.Stack} The created CDK stack.
+   */
+  private createStack(ctx: TransformerContextProvider, stackName: string): cdk.Stack {
+    const stack = ctx.stackManager.createStack(stackName);
+    stack.templateOptions.templateFormatVersion = '2010-09-09';
+    stack.templateOptions.description = 'An auto-generated nested stack for the @generation directive.';
+    return stack;
+  }
+
+  /**
+   * Creates a pipeline resolver for the Generation transformer.
+   * @param {TransformerContextProvider} ctx - The transformer context provider.
+   * @param {string} parentName - The name of the parent type.
+   * @param {string} fieldName - The name of the field.
+   * @param {string} resolverResourceId - The ID for the resolver resource.
+   * @param {MappingTemplateProvider} invokeBedrockFunction - The invoke Bedrock function.
+   */
+  private createPipelineResolver(
+    ctx: TransformerContextProvider,
+    parentName: string,
+    fieldName: string,
+    resolverResourceId: string,
+    invokeBedrockFunction: { req: MappingTemplateProvider; res: MappingTemplateProvider },
+    dataSource: cdk.aws_appsync.HttpDataSource,
+  ): void {
+    const conversationPipelineResolver = new TransformerResolver(
+      parentName,
+      fieldName,
+      resolverResourceId,
+      invokeBedrockFunction.req,
+      invokeBedrockFunction.res,
+      ['auth'],
+      [],
+      dataSource as any,
+      { name: 'APPSYNC_JS', runtimeVersion: '1.0.0' },
```

Comment:
Probably should eventually be a constant defined in amplify-graphql-transformer-core, but it's only being used here for now. Will add one in a follow up if / when it becomes necessary.